### ISSUE TITLE
SiteNow Events view page

### ISSUE THREAD

**Issue by joewhitsitt** [uiowa/uiowa#3171]
Split from #2396
Follow-up to #3170
Blocked by #3458

# Story

As a site owner with the local events content type split enabled, I would like to modify aspects of the view without asking a developer.

The problem right now with providing this solution is that being a feature split and not part of the default, the view needs to be imported when the feature is enabled but ignored when the configuration page tries to ignore it. This requires a dedicated module to install/uninstall. To do this right, it is my opinion that #3458 should be addressed first so that sitenow_events can be used.

# Proposed solution

- [ ] SiteNow Events configuration page similar to SiteNow articles that manipulates views.view.events with options to enable/disable view, custom title, custom header content.
- [ ] Config ignore views.view.events within the feature split.
- [ ] Install/Uninstall logic within sitenow_events that imports/removes the view configuration.

**Comment by joewhitsitt** [comment]
There is a separate issue related to this that would tackle these differently going forward. https://github.com/uiowa/uiowa/issues/3151

IMO, I think we should keep with these dedicated view pages that can be disabled by the webmaster.

**Comment by bspeare** [comment]
I don't know whether this is in the scope of this particular issue, but we might want to look into styling this view of events like the screenshot here: https://github.com/uiowa/uiowa/issues/2727#issuecomment-834403434.

**Comment by joewhitsitt** [comment]
I think the custom path requirement should be removed. I had troubles with ignoring this path at the feature split level which is different than the default article/person paths. That and I think _if_ local events are used, they should have consistent paths. A developer implementing the feature split can split the path config if changes are needed.

**Comment by joewhitsitt** [comment]
Based on the conversation at the Drupal meeting I think the view should be hardcoded as config within the feature split. Admissions should then site split the view for their needs.

We then need to prioritize https://github.com/uiowa/uiowa/issues/3458 so that the feature split can be served by a dedicated module that will handle the custom functionality better.

**Comment by joewhitsitt** [comment]
https://github.com/uiowa/uiowa/issues/3151 is still something to consider but has other problems to solve.

**Comment by joewhitsitt** [comment]
So even though #3513 was closed, I would like to hijack this issue and update it for when we can tackle and provide this functionality.

**Comment by joewhitsitt** [comment]
To move forward with this for Admissions based on recent discussions, we should just create this view for admissions only without configuration. https://github.com/uiowa/uiowa/issues/3458#issuecomment-870836241

https://github.com/uiowa/uiowa/issues/3773

**Comment by joewhitsitt** [comment]
Admissions got their view page. Not sure we want/need to pursue this. Events block should be able to handle most use-cases and this is still tricky to accomplish like articles/people if not part of the default.

**Comment by joewhitsitt** [comment]
There is an always on /events view page with the default event feature with configurable filters. Additional work could be done to modify other parts of the view on the fly without needing to config_ignore it.

**Comment by joewhitsitt** [comment]
Closing. As noted in https://github.com/uiowa/uiowa/issues/3171#issuecomment-1648039840. Functionality already exists minus a couple details. As those features are requested a new issue can be created.
